server/src/main/java/org/elasticsearch/index/refresh/RefreshStats.java

```diff
@@ -129,6 +160,27 @@ public int getListeners() {
         return listeners;
     }
 
+    /**
+     * Timestamp of the last refresh.
+     */
+    public long getLastRefreshTime() {
+        return lastRefreshTime;
+    }
+
+    /**
+     * Timestamp of the last external refresh.
+     */
+    public long getLastExternalRefreshTime() {
+        return lastExternalRefreshTime;
+    }
+
+    /**
+     * Whether there are changes that need to be written to disk or not.
+     */
+    public boolean getHasUnwrittenChanges() {
```

Comment:
I went back and forth between `unwritten` and `unrefreshed` and settled on `unwritten`, but I'm happy to change it to `unrefreshed`..!